src/documentation/document_index.ipynb

```diff
@@ -291,6 +291,223 @@
+    "The DocumentIndex also supports filter-indexes, which gives us the ability to provide specific filters in case we want to filter our search based on each document's metadata.\n",
+    "\n",
+    "To do so, let's first upload another version of our documents but this time with some metadata e.g. the \"title\" field."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "for doc in documents:\n",
+    "    document_path = DocumentPath(\n",
+    "        collection_path=collection_path, document_name=doc[\"name\"]\n",
+    "    )\n",
+    "    document_index.add_document(\n",
+    "        document_path,\n",
+    "        contents=DocumentContents(\n",
+    "            contents=[doc[\"content\"]], metadata={\"title\": doc[\"name\"]}\n",
+    "        ),\n",
+    "    )"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Preparation of the Filter Index\n",
+    "To be able to use metadata filtering, we need to first check the following:\n",
+    "1. Check if we already have a search index assigned. If not, we need to assign one because filter-indexes can be defined at the namespace level but can only be assigned to already existing search indexes \n",
+    "2. Define a new filter-index configuration for our specific collection metadata.\n",
+    "3. Assign the filter-index that we created to a search index."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# 1\n",
+    "# list all the assigned search indexes for our collection\n",
+    "document_index.list_assigned_index_names(collection_path=collection_path)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# 2\n",
+    "# define a new filter-index\n",
+    "document_index.create_filter_index_in_namespace(\n",
+    "    namespace=collection_path.namespace,\n",
+    "    filter_index_name=\"demo-title\",  # this is how our filter-index is identified in our namespace\n",
+    "    field_name=\"title\",  # this is the name of the field to which we want to apply our filter\n",
+    "    field_type=\"string\",  # type of the field we want to apply our filter to. Must be one of \"string\", \"integer\", \"float\", \"boolean\" or \"datetime\"\n",
+    ")\n",
+    "\n",
+    "# let's check if our index is present now\n",
+    "\"demo-title\" in document_index.list_filter_indexes_in_namespace(\n",
+    "    namespace=collection_path.namespace\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# 3\n",
+    "# assign our new filter-index to our collection\n",
+    "document_index.assign_filter_index_to_search_index(\n",
+    "    collection_path=collection_path,\n",
+    "    index_name=\"asym-128\",  # we assign it to asym-128\n",
```

Comment:
If we are using `intelligence-layer-sdk-demo-collection` as a collection as assigned in the third cell, there does not seem to be an `asym-128` index. We could use `intelligence-layer-sdk-demo-index` instead.